modules/3d/src/pointcloud/coders_obj.cpp

```diff
@@ -0,0 +1,114 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "coders_obj.hpp"
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <opencv2/core/utils/logger.hpp>
+#include "utils.hpp"
+
+namespace cv
+{
+namespace pc
+{
+
+void ObjDecoder::readData(std::vector<Point3f> &points, std::vector<Point3f> &normals, std::vector<std::vector<int32_t>> &indices)
+{
+    points.clear();
+    normals.clear();
+    indices.clear();
+
+    std::ifstream file(m_filename, std::ios::binary);
+    if (!file)
+    {
+        CV_Error(Error::StsError, "Impossible to open the file !\n");
+    }
+    std::string s;
+
+    std::vector<std::string> unsupported_keys;
```

Comment:
use `std::unordered_set<std::string>` instead. and make it static member of the class